drift_dev/test/backends/build/build_integration_test.dart

```diff
@@ -614,6 +616,153 @@ class MyDatabase {
+
+  BlobColumn get profilePicture => blob()();
+  DateTimeColumn get creationTime => dateTime()
+  // ignore: recursive_getters
+      .check(creationTime.isBiggerThan(Constant(DateTime.utc(1950))))
+      .withDefault(currentDateAndTime)();
+}
+
+@DataClassName('Category')
+class Categories extends Table with AutoIncrement {
+  TextColumn get description =>
+      text().named('desc').customConstraint('NOT NULL UNIQUE')();
+  IntColumn get priority =>
+      intEnum<CategoryPriority>().withDefault(const Constant(0))();
+
+  TextColumn get descriptionInUpperCase =>
+      text().generatedAs(description.upper())();
+}
+
+enum CategoryPriority { low, medium, high }
+abstract class CategoryTodoCountView extends View {
+  TodosTable get todos;
+  Categories get categories;
+
+  Expression<int> get categoryId => categories.id;
+  Expression<String> get description =>
+      categories.description + const Variable('!');
+  Expression<int> get itemCount => todos.id.count();
+
+  @override
+  Query as() => select([categoryId, description, itemCount])
+      .from(categories)
+      .join([innerJoin(todos, todos.category.equalsExp(categories.id))])
+    ..groupBy([categories.id]);
+}
+abstract class ComboGroupView extends View {
+  late final DatastoreDb attachedDatabase;
+  IntColumn get comboGroupID => attachedDatabase.comboGroup.comboGroupID;
+  IntColumn get objectNumber => attachedDatabase.comboGroup.objectNumber;
+  TextColumn get stringText => attachedDatabase.stringTable.stringText;
+  // ComboGroup get comboGroup => attachedDatabase.comboGroup;
+  // late final ComboGroup comboGroup;
+  @override
+  Query as() => select([
+    comboGroupID,
+    objectNumber,
+    stringText,
+  ]).from(attachedDatabase.comboGroup).join([
+    innerJoin(
+        attachedDatabase.stringTable,
+        attachedDatabase.stringTable.stringNumberID
+            .equalsExp(attachedDatabase.comboGroup.nameID)),
+  ]);
+}
+
+@DriftDatabase(
+  tables: [TodosTable, Categories],
+  include: {'combo_group.drift','string_table.drift'},
+  views: [CategoryTodoCountView,ComboGroupView],
+)
+class DatastoreDb extends _\$DatastoreDb {
+  DatastoreDb(super.e);
+  @override
+  int get schemaVersion => 1;
+}
+        ''',
+          'a|lib/drift/combo_group.drift': '''
+        CREATE TABLE [COMBO_GROUP](
+	[ComboGroupID] [int] NOT NULL PRIMARY KEY,
+	[HierStrucID] [bigint] NULL,
+	[ObjectNumber] [int] NULL,
+	[NameID] [bigint] NULL,
+	[OptionBits] [nvarchar](8) NULL,
+	[SluIndex] [int] NULL,
+	[HhtSluIndex] [int] NULL);
+	''',
+          'a|lib/drift/string_table.drift': '''
+CREATE TABLE [STRING_TABLE](
+	[StringID] [bigint] NOT NULL PRIMARY KEY,
+	[StringNumberID] [bigint] NULL,
+	[LangID] [int] NULL,
+	[IsVisible] [bit] NOT NULL DEFAULT ((1)),
+	[IsDeleted] [bit] NOT NULL DEFAULT ((0)),
+	[HierStrucID] [bigint] NULL,
+	[PosRef] [bigint] NULL,
+	[StringText] [nvarchar](128) NULL
+);
+	''',
+        },
+        modularBuild: true,
```

Comment:
Wouldn't this actually generate invalid code in the modular build, since that tries to infer imports which isn't possible with this? Let's disable this for this test, that also avoids having to test for the two other outputs.

Reply:
Done, it did work but we don't require modularBuild